scripts/update-rate-card-rates/update-rate-card-rates.py

```diff
@@ -0,0 +1,160 @@
+from datetime import datetime, timedelta, timezone
+from pprint import pprint
+import requests
+import argparse
+import os
+import csv
+import random
+import uuid
+import requests
+from dotenv import load_dotenv
+from collections import defaultdict
+import json
+
+load_dotenv()  # take environment variables from .env.
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--csv_file', type=str, help='CSV file with rate data')
+
+token = os.getenv('API_TOKEN')
+if not token:
+    raise ValueError("Set API_TOKEN environment variable to metronome API key")
+
+args = parser.parse_args()
+csv_file_path = args.csv_file
+if not csv_file_path:
+    raise ValueError("csv_file must be set")
+
+api_url = os.getenv('API_URL', 'https://api.metronome.com')
+
+headers = {
+    "Authorization": f"Bearer {token}",
+    "Content-Type": "application/json"
+}
+
+# read all products and their current custom fields / tags
+productHash = {}
+next_page = None
+print('retrieving current products')
+while True:
+    next_page_param = f"&next_page={next_page}" if next_page else ""
+    response = requests.post(
+        f"{api_url}/v1/contract-pricing/products/list?limit=100{next_page_param}",
+        headers=headers,
+        json={}
+    )
+    response.raise_for_status()
+
+    response_json = response.json()
+    response_data = response_json['data']
+    for product in response_data:
+        current = product['current']
+        productHash[current['name']] = {
+            'id': product['id'],
+        }
+
+    next_page = response_json['next_page']
+
+    if not next_page:
+        break
+
+rateCardHash = {}
+next_page = None
+print('retrieving current rate cards')
+while True:
+    next_page_param = f"&next_page={next_page}" if next_page else ""
+    response = requests.post(
+        f"{api_url}/v1/contract-pricing/rate-cards/list?limit=100{next_page_param}",
+        headers=headers,
+        json={}
+    )
+    response.raise_for_status()
+
+    response_json = response.json()
+    response_data = response_json['data']
+    for rate_card in response_data:
+        rateCardHash[rate_card['name']] = {
+            'id': rate_card['id'],
+        }
+
+    next_page = response_json['next_page']
+
+    if not next_page:
+        break
+
+
+rates_to_send_by_rate_card = defaultdict(list)
+with open(csv_file_path) as csv_file:
+    csv_reader = csv.DictReader(csv_file, delimiter=',')
+
+    for row in csv_reader:
+        rate_card_name = row['rate_card_name']
```

Comment:
should this use the rate card name or alias?

Reply:
name for now, but I could easily see supporting alias here in the future